--- src/zsolve/Lattice.hpp.orig	2008-04-03 15:16:04.000000000 +0200
+++ src/zsolve/Lattice.hpp	2013-05-03 09:35:54.997601358 +0200
@@ -96,6 +96,10 @@ public:
     {
         for (size_t column = 0; column < VectorArray <T> :: m_variables; column++)
         {
+            // Abort if there are not enough rows.
+            if (column < VectorArray<T>:: m_vectors)
+              break;
+
             int current_index;
             T current_value;
             int best_index = column;
